src/components/select/option/option.test.tsx

```diff
@@ -0,0 +1,253 @@
+    render(
+      <ul>
+        <Option text="foo" value="1" disabled />
+      </ul>
+    );
+    const option = screen.getByRole("option", { name: "foo" });
+
+    expect(option).toHaveStyleRule("color", "var(--colorsUtilityYin030)");
+    expect(option).toHaveStyle("cursor: not-allowed");
+  });
+
+  it("should set the aria-disabled attribute to true", () => {
+    render(
+      <ul>
+        <Option text="foo" value="1" disabled />
+      </ul>
+    );
+    const option = screen.getByRole("option", { name: "foo" });
+
+    expect(option).toHaveAttribute("aria-disabled", "true");
+  });
+
+  it("should not call `onSelect` or `onClick` when the user clicks on the element", async () => {
+    const user = userEvent.setup();
+    const onSelect = jest.fn();
+    const onClick = jest.fn();
+    render(
+      <ul>
+        <Option
+          text="foo"
+          value="1"
+          onClick={onClick}
+          onSelect={onSelect}
+          disabled
+        />
+      </ul>
+    );
+
+    await user.click(screen.getByRole("option", { name: "foo" }));
+
+    expect(onSelect).not.toHaveBeenCalled();
+    expect(onClick).not.toHaveBeenCalled();
+  });
+});
+
+describe("when `disabled` prop is not set", () => {
+  it("should have expected style", () => {
+    render(
+      <ul>
+        <Option text="foo" value="1" />
+      </ul>
+    );
+    const option = screen.getByRole("option", { name: "foo" });
+
+    expect(option).not.toHaveStyleRule("color", "var(--colorsUtilityYin030)");
+    expect(option).toHaveStyle("cursor: pointer");
+  });
+
+  it("should not set the aria-disabled attribute", () => {
+    render(
+      <ul>
+        <Option text="foo" value="1" />
+      </ul>
+    );
+    const option = screen.getByRole("option", { name: "foo" });
+
+    expect(option).not.toHaveAttribute("aria-disabled");
+  });
+
+  it("should call `onSelect` and `onClick` when the user clicks on the element", async () => {
+    const user = userEvent.setup();
+    const onSelect = jest.fn();
+    const onClick = jest.fn();
+    render(
+      <ul>
+        <Option text="foo" value="1" onClick={onClick} onSelect={onSelect} />
+      </ul>
+    );
+
+    await user.click(screen.getByRole("option", { name: "foo" }));
+
+    expect(onSelect).toHaveBeenCalled();
+    expect(onClick).toHaveBeenCalled();
+  });
+
+  it("should call `onSelect` when the user clicks on the element and `onClick` is not set but custom `id` passed", async () => {
+    const user = userEvent.setup();
+    const onSelect = jest.fn();
+    render(
+      <ul>
+        <Option text="foo" value="bar" onSelect={onSelect} />
+      </ul>
+    );
+
+    await user.click(screen.getByRole("option", { name: "foo" }));
+
+    expect(onSelect).toHaveBeenCalledWith({
+      text: "foo",
+      value: "bar",
+      id: mockedGuid,
+    });
+  });
+
+  it("should call `onSelect` when the user clicks on the element and `onClick` is not set but custom `id` is passed", async () => {
```

Comment:
**suggestion(non-blocking)**: similarly, what do you think about this tweak for clarity?:

```suggestion
  it("calls `onSelect` when clicked, without `onClick` set, but with `id` provided.", async () => {
```